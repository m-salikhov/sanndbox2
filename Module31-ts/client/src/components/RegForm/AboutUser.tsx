import React, { useState } from 'react';
import axios from 'axios';

export default function AboutUser() {
	const [name, setName] = useState('');
	const [bdayDate, setBdayDate] = useState('');
	const [email, setEmail] = useState('');
	const [phone, setPhone] = useState('');
	const [passport, setPassport] = useState('');
	const [passDate, setPassDate] = useState('');
	const [passOrg, setPassOrg] = useState('');
	const [passOrgCode, setPassOrgCode] = useState('');
	const [licenseNumber, setLicenseNumber] = useState('');
	const [dateLicense, setDateLicense] = useState('');
	const [pass, setPass] = useState('');
	const [passRepeat, setPassRepeat] = useState('');

	const [message, setMessage]: any = useState('');

	function onSubmit(e: any) {
		e.preventDefault();
		axios
			.post('/users', {
				name,
				bdayDate,
				email,
				phone,
				passport,
				passDate,
				passOrg,
				passOrgCode,
				licenseNumber,
				dateLicense,
				pass,
				passRepeat,
			})
			.then((res) => res.data)
			.then((data) => setMessage(data));
	}

	return (
		<>
			<form>
				<h2>Информация о вас</h2>
				<label>
					ФИО{' '}
					<input
						className='name'
						value={name}
						onChange={(e) => setName(e.target.value)}
						placeholder='ФИО полностью'
						type='text'
					/>
				</label>
				<label>
					Дата рождения
					<div className='date'>
						<input
							className='dateBirthday'
							value={bdayDate}
							onChange={(e) => setBdayDate(e.target.value)}
							type='date'
						/>
					</div>
				</label>
				<label>
					Электронная почта
					<input
						className='email'
						placeholder='mail@example.com'
						value={email}
						onChange={(e) => setEmail(e.target.value)}
						type='email'
					/>
				</label>
				<label>
					Телефон
					<input
						className='phone'
						placeholder='+7 900 000-00-00'
						value={phone}
						onChange={(e) => setPhone(e.target.value)}
						type='tel'
					/>
				</label>
				<h2>Паспорт</h2>
				<label>
					Серия и номер
					<input
						className='series'
						placeholder='0000 000000'
						value={passport}
						onChange={(e) => setPassport(e.target.value)}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='datePassport'
							value={passDate}
							onChange={(e) => setPassDate(e.target.value)}
							type='date'
						/>
					</div>
				</label>
				<label>
					Кем выдан
					<input
						className='issued'
						placeholder='Название органа выдавшего паспорт'
						value={passOrg}
						onChange={(e) => setPassOrg(e.target.value)}
						type='text'
					/>
				</label>
				<label>
					Код подразделения
					<input
						className='codeIssued'
						placeholder='000-000'
						value={passOrgCode}
						onChange={(e) => setPassOrgCode(e.target.value)}
						type='text'
					/>
				</label>
				<h2>Водительское удостоверение </h2>
				<label>
					Серия и номер
					<input
						className='licenseNumber'
						placeholder='0000 000000'
						value={licenseNumber}
						onChange={(e) => setLicenseNumber(e.target.value)}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='dateLicense'
							value={dateLicense}
							onChange={(e) => setDateLicense(e.target.value)}
							type='date'
						/>
					</div>
				</label>
				<h2>Пароль</h2>
				<label>
					Придумайте пароль
					<input
						className='pass'
						value={pass}
						onChange={(e) => setPass(e.target.value)}
						type='password'
					/>
				</label>
				<label>
					Повторите пароль{' '}
					<input
						className='passRepeat'
						value={passRepeat}
						onChange={(e) => setPassRepeat(e.target.value)}
						type='password'
					/>
				</label>
				<div className='rectangle'>
					<h2 style={{ color: 'red' }}>{message.name}</h2>
					<button className='subButton' type='submit' onClick={onSubmit}>
						Продолжить
					</button>
				</div>
			</form>
		</>
	);
}
