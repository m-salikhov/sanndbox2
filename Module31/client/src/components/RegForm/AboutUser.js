import { useMutation } from '@apollo/client';
import { useState } from 'react';
import { CREATE_USER_REQUEST } from '../../utils/gql-req';

export default function AboutUser() {
	const [inputUser, setInputUser] = useState({
		username: '',
		bdayDate: '',
		email: '',
		phone: '',
		passport: '',
		passDate: '',
		passOrg: '',
		passOrgCode: '',
		licenseNumber: '',
		dateLicense: '',
		pass: '',
		passRepeat: '',
	});

	const [message, setMessage] = useState('');

	const onChange = (e) => {
		setInputUser({ ...inputUser, [e.target.name]: e.target.value });
	};

	const [createNewUser] = useMutation(CREATE_USER_REQUEST, {
		variables: { userInput: inputUser },
	});

	console.log(inputUser);

	const onSubmit = (event) => {
		event.preventDefault();
		if (inputUser.pass !== inputUser.passRepeat)
			return setMessage('Пароль не совпадает');
		createNewUser();
	};

	return (
		<>
			<form onSubmit={onSubmit}>
				<h2>Информация о вас</h2>
				<label>
					ФИО{' '}
					<input
						className='name'
						name='username'
						onChange={onChange}
						value={inputUser.username}
						placeholder='ФИО полностью'
						type='text'
					/>
				</label>
				<label>
					Дата рождения
					<div className='date'>
						<input
							className='dateBirthday'
							name='bdayDate'
							onChange={onChange}
							value={inputUser.bdayDate}
							type='date'
						/>
					</div>
				</label>
				<label>
					Электронная почта
					<input
						className='email'
						placeholder='mail@example.com'
						name='email'
						onChange={onChange}
						value={inputUser.email}
						type='email'
					/>
				</label>
				<label>
					Телефон
					<input
						className='phone'
						placeholder='+7 900 000-00-00'
						name='phone'
						onChange={onChange}
						value={inputUser.phone}
						type='tel'
					/>
				</label>
				<h2>Паспорт</h2>
				<label>
					Серия и номер
					<input
						className='series'
						placeholder='0000 000000'
						name='passport'
						onChange={onChange}
						value={inputUser.passport}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='datePassport'
							name='passDate'
							onChange={onChange}
							value={inputUser.passDate}
							type='date'
						/>
					</div>
				</label>
				<label>
					Кем выдан
					<input
						className='issued'
						placeholder='Название органа выдавшего паспорт'
						name='passOrg'
						onChange={onChange}
						value={inputUser.passOrg}
						type='text'
					/>
				</label>
				<label>
					Код подразделения
					<input
						className='codeIssued'
						placeholder='000-000'
						name='passOrgCode'
						onChange={onChange}
						value={inputUser.passOrgCode}
						type='text'
					/>
				</label>
				<h2>Водительское удостоверение </h2>
				<label>
					Серия и номер
					<input
						className='licenseNumber'
						placeholder='0000 000000'
						name='licenseNumber'
						onChange={onChange}
						value={inputUser.licenseNumber}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='dateLicense'
							name='dateLicense'
							onChange={onChange}
							value={inputUser.dateLicense}
							type='date'
						/>
					</div>
				</label>
				<h2>Пароль</h2>
				<label>
					Придумайте пароль
					<input
						className='pass'
						name='pass'
						onChange={onChange}
						value={inputUser.pass}
						type='password'
					/>
				</label>
				<label>
					Повторите пароль{' '}
					<input
						className='passRepeat'
						name='passRepeat'
						onChange={onChange}
						value={inputUser.passRepeat}
						type='password'
					/>
				</label>
				<div className='rectangle'>
					<h2 style={{ color: 'red' }}>{message}</h2>
					<button className='subButton' type='submit'>
						Продолжить
					</button>
				</div>
			</form>
		</>
	);
}
