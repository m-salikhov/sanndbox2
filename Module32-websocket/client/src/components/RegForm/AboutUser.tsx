import { useState } from 'react';
import axios from 'axios';

export default function AboutUser() {
	const [inputUser, setInputUser] = useState({
		username: '',
		bdayDate: '',
		email: '',
		phone: '',
		passport: '',
		passDate: '',
		passOrg: '',
		passOrgCode: '',
		licenseNumber: '',
		dateLicense: '',
		pass: '',
	});

	const [passRepeat, setPassRepeat] = useState('');

	const onChange = (e: any) => {
		setInputUser({ ...inputUser, [e.target.name]: e.target.value });
	};

	const [message, setMessage] = useState('');

	function onSubmit(e: { preventDefault: () => void }) {
		e.preventDefault();
		if (inputUser.pass !== passRepeat) return setMessage('Пароль не совпадает');
		axios
			.post('/users', inputUser)
			.then((res) => res.data)
			.then((data) => setMessage(data.username))
			.catch((error) => setMessage(error.response.data.message));
	}

	console.log(message);
	return (
		<>
			<form onSubmit={onSubmit}>
				<h2>Информация о вас</h2>
				<label>
					ФИО{' '}
					<input
						className='name'
						name='username'
						onChange={onChange}
						value={inputUser.username}
						placeholder='ФИО полностью'
						type='text'
					/>
				</label>
				<label>
					Дата рождения
					<div className='date'>
						<input
							className='dateBirthday'
							name='bdayDate'
							onChange={onChange}
							value={inputUser.bdayDate}
							type='date'
						/>
					</div>
				</label>
				<label>
					Электронная почта
					<input
						className='email'
						placeholder='mail@example.com'
						name='email'
						onChange={onChange}
						value={inputUser.email}
						type='email'
					/>
				</label>
				<label>
					Телефон
					<input
						className='phone'
						placeholder='+7 900 000-00-00'
						name='phone'
						onChange={onChange}
						value={inputUser.phone}
						type='tel'
					/>
				</label>
				<h2>Паспорт</h2>
				<label>
					Серия и номер
					<input
						className='series'
						placeholder='0000 000000'
						name='passport'
						onChange={onChange}
						value={inputUser.passport}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='datePassport'
							name='passDate'
							onChange={onChange}
							value={inputUser.passDate}
							type='date'
						/>
					</div>
				</label>
				<label>
					Кем выдан
					<input
						className='issued'
						placeholder='Название органа выдавшего паспорт'
						name='passOrg'
						onChange={onChange}
						value={inputUser.passOrg}
						type='text'
					/>
				</label>
				<label>
					Код подразделения
					<input
						className='codeIssued'
						placeholder='000-000'
						name='passOrgCode'
						onChange={onChange}
						value={inputUser.passOrgCode}
						type='text'
					/>
				</label>
				<h2>Водительское удостоверение </h2>
				<label>
					Серия и номер
					<input
						className='licenseNumber'
						placeholder='0000 000000'
						name='licenseNumber'
						onChange={onChange}
						value={inputUser.licenseNumber}
						type='text'
					/>
				</label>
				<label>
					Дата выдачи
					<div className='date'>
						<input
							className='dateLicense'
							name='dateLicense'
							onChange={onChange}
							value={inputUser.dateLicense}
							type='date'
						/>
					</div>
				</label>
				<h2>Пароль</h2>
				<label>
					Придумайте пароль
					<input
						className='pass'
						name='pass'
						onChange={onChange}
						value={inputUser.pass}
						type='password'
					/>
				</label>
				<label>
					Повторите пароль{' '}
					<input
						className='passRepeat'
						name='passRepeat'
						onChange={(e) => setPassRepeat(e.target.value)}
						value={passRepeat}
						type='password'
					/>
				</label>
				<div className='rectangle'>
					<h2 style={{ color: 'red' }}>{message}</h2>
					<button className='subButton' type='submit'>
						Продолжить
					</button>
				</div>
			</form>
		</>
	);
}
