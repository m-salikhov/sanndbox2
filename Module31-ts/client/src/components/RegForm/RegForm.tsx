import React from 'react';
import AboutUser from './AboutUser';
import '../../styles/regform.scss';

export default function RegForm() {
	return (
		<main className='reg-form'>
			<p>шаг 1 из 3</p>
			<h1>Расскажите о себе</h1>
			<div className='container'>
				<AboutUser />
			</div>
		</main>
	);
}
