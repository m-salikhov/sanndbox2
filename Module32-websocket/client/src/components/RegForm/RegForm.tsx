import AboutUser from './AboutUser';
import '../../styles/regform.scss';

export default function RegForm() {
	return (
		<main className='reg-form'>
			<p>шаг 1 из 3</p>
			<h1 className='title'>Расскажите о себе</h1>
			<div className='container'>
				<AboutUser />
			</div>
		</main>
	);
}
