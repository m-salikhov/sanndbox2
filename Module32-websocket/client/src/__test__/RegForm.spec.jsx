import { shallow } from 'enzyme';
import RegForm from '../components/RegForm/RegForm';

describe('RegForm component', () => {
	it('should render RegForm component', () => {
		const component = shallow(<RegForm />);
		expect(component).toMatchSnapshot();
	});
});
