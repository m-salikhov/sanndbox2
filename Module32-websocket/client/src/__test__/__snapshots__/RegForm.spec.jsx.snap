// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegForm component should render RegForm component 1`] = `
<main
  className="reg-form"
>
  <p>
    шаг 1 из 3
  </p>
  <h1
    className="title"
  >
    Расскажите о себе
  </h1>
  <div
    className="container"
  >
    <AboutUser />
  </div>
</main>
`;
